test/unit/math/test_ad.hpp

```diff
@@ -1432,6 +1433,78 @@ void expect_ad_vectorized_binary(const F& f, const T1& x, const T2& y) {
   expect_ad_vectorized_binary(tols, f, x, y);
 }
 
+template <typename T1, typename T2,
+          require_all_stan_scalar_t<T1, T2>* = nullptr>
+void expect_float_equal(const T1& x, const T2& y) {
+  EXPECT_FLOAT_EQ(x.val(), y.val());
+  EXPECT_FLOAT_EQ(x.adj(), y.adj());
+}
+template <
+    typename T1, typename T2,
+    require_any_eigen_t<T1, value_type_t<T1>, T2, value_type_t<T2>>* = nullptr>
+void expect_float_equal(const T1& x, const T2& y) {
+  EXPECT_MATRIX_FLOAT_EQ(x.val(), y.val());
+  EXPECT_MATRIX_FLOAT_EQ(x.adj(), y.adj());
+}
+
+/**
+ * For an unary function check that an Eigen matrix of vars and a var with an
+ * inner eigen type return the same values and adjoints. This is done by
+ * calling the function on both types, summing the results, then taking the
+ * gradient of the sum which will propogate the adjoints upwards.
+ *
+ * @tparam F A lambda or functor type that calls the unary function.
+ * @tparam T An Eigen matrix type
+ * @param f a lambda
+ * @param x An Eigen matrix.
+ *
+ */
+template <typename F, typename T>
+void expect_ad_matvar(F&& f, const T& x) {
```

Comment:
This should take in a tolerances argument to support `expect_near_rel`: https://github.com/stan-dev/math/blob/0f9b5f031e2171994a49d9dce95f6c50e857066a/test/unit/math/test_ad.hpp#L1123